### PR TITLE
Improve: ws before colon for constraint

### PR DESCRIPTION
**Pull request by hhugo**
following comments in #284

### CODE REVIEW

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -1518,7 +1517,7 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext ({ast= exp} as xexp)
   | Pexp_constraint (e, t) ->
       wrap_fits_breaks "(" ")"
         ( fmt_expression c (sub_exp ~ctx e)
-        $ fmt "@ : "
+        $ fmt "@;<1 2>: "
```
Comment (jberdine) [inline src/Fmt_ast.ml]: I suspect that the whole body of the Pexp_constraint case should be wrapped in a box a la Ppat_constraint above, and `@;<1 2>` could go back to `@ `

Comment (hhugo) [reply src/Fmt_ast.ml]: done

File: src/Fmt_ast.ml [hunk 2]
```diff
@@ -2277,7 +2277,7 @@ and fmt_class_expr c ?eol ?(box= true) ({ast= exp} as xexp) =
   | Pcl_constraint (e, t) ->
       wrap_fits_breaks "(" ")"
         ( fmt_class_expr c (sub_cl ~ctx e)
-        $ fmt "@ : "
+        $ fmt "@;<1 2>: "
```
Comment (jberdine) [inline src/Fmt_ast.ml]: add box as above?

Comment (hhugo) [reply src/Fmt_ast.ml]: yes

**Pull request merged**